### DISCUSSION TITLE
IM cannot open old tiff-files

### DISCUSSION THREAD

**Discussion by DanielDrs** [ImageMagick/ImageMagick#5957] [Digital Image Processing]
Hallo,

IM cannot read some old tiff-files written by RawTherapee.    Gimp and Photoshop read these files without problems. 
When I open and safe such a file in Gimp or Photoshop, IM can read the output file.

Some files are under    https://www.dropbox.com/sh/rfnbmekev5gzy9c/AADwmwbiDYBWXNDcQCAjzUSQa?dl=0
orig.tif    is such an old problem file.
ps.tif  is the result from opening and saving orig.tif in Photoshop.
gmp.tif   is the result from opening and saving orig.tif in Gimp.

magick orig.tif  test.jpg
magick: Sanity check on directory count failed, this is probably not a valid IFD offset. `TIFFFetchDirectory' @ error/tiff.c/TIFFErrors/661.
magick: Failed to read custom directory at offset 8. `TIFFReadCustomDirectory' @ error/tiff.c/TIFFErrors/661.

I am looking for a method to read or repair these old files in a batch.....

Best regards, Daniel

magick -version
Version: ImageMagick 7.0.10-27 Q16 x86_64 2021-02-07 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): bzlib cairo djvu fftw fontconfig fpx freetype gslib gvc jbig jng jp2 jpeg lcms lqr lzma pangocairo png ps raqm raw rsvg tiff webp x xml zlib

**Comment by fmw42** [comment]
EXIFTOOL says

```
Warning:
Bad format (21353) for ExifIFD entry 0
```



Imagemagick identify -verbose says:

```
identify: Sanity check on directory count failed, this is probably not a valid IFD offset. `TIFFFetchDirectory' @ error/tiff.c/TIFFErrors/598.
identify: Failed to read custom directory at offset 8. `TIFFReadCustomDirectory' @ error/tiff.c/TIFFErrors/598.
```

So the file is likely malformed to the TIFF specifications.

**Comment by DanielDrs** [comment]
Thanks.  I found a solution:

`exiftool -all= filename.tif   `

repairs the file.

Daniel
